apps/definition_extract/lib/extract.ex

```diff
@@ -8,22 +8,20 @@ defmodule Extract do
           id: uuid,
           dataset_id: String.t(),
           subset_id: String.t(),
-          destination: String.t(),
-          steps: [Extract.Step.t()],
-          dictionary: Dictionary.t(),
-          message_key: list,
-          config: map
+          source: Source.t(),
+          decoder: Decoder.t(),
+          destination: Destination.t(),
+          dictionary: Dictionary.t()
```

Comment:
Wait, so we no longer have steps? I was thinking that steps would include a step that includes a source (struct) etc. But are we actually removing those now? (Maybe steps are part of source / dest now, I need to keep reading).

I believe this means that define will need to create code / a component for each type of source struct, as opposed to a more generic 'step' component? May actually make the front end code cleaner, but also more verbose. Would be curious what Natty/Nick's thoughts are.

Reply:
So my current thought was the extraction pipeline( the steps) becomes one implementation of a source.  So we are not getting rid of steps but moving them to a source implementation.  However the idea of reversing it and making a source of of the steps is really interesting.  I would love to brainstorm around this today or Monday.

Reply:
I'd like that!